src/main/java/gift/repository/ProductRepository.java

```diff
@@ -0,0 +1,77 @@
+package gift.repository;
+
+import gift.entity.Product;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.simple.SimpleJdbcInsert;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class ProductRepository implements ProductRepositoryInterface {
```

Comment:
```suggestion
@Repository
public class ProductJdbcRepository implements ProductRepository {
```
- 외와 같이 명명하는 것은 어떨까요? 인터페이스는 행위를 정의하고, 그 구현부에 특성을 표기해주는게 더 자연스러워보입니다.